src/Controls/src/Core/Shell/ShellItem.cs

```diff
@@ -222,10 +222,34 @@ internal static ShellItem CreateFromShellSection(ShellSection shellSection)
 			result.Route = Routing.GenerateImplicitRoute(shellSection.Route);
 
 			result.Items.Add(shellSection);
-			result.SetBinding(TitleProperty, new Binding(nameof(Title), BindingMode.OneWay, source: shellSection));
-			result.SetBinding(IconProperty, new Binding(nameof(Icon), BindingMode.OneWay, source: shellSection));
-			result.SetBinding(FlyoutDisplayOptionsProperty, new Binding(nameof(FlyoutDisplayOptions), BindingMode.OneTime, source: shellSection));
-			result.SetBinding(FlyoutIconProperty, new Binding(nameof(FlyoutIcon), BindingMode.OneWay, source: shellSection));
+			result.SetBinding(
+				TitleProperty,
+				TypedBinding.ForSingleNestingLevel(
+					nameof(ShellSection.Title),
+					static (ShellSection section) => section.Title,
+					mode: BindingMode.OneWay,
+					source: shellSection));
+			result.SetBinding(
+				IconProperty,
+				TypedBinding.ForSingleNestingLevel(
+					nameof(ShellSection.Icon),
+					static (ShellSection section) => section.Icon,
+					mode: BindingMode.OneWay,
+					source: shellSection));
+			result.SetBinding(
+				FlyoutDisplayOptionsProperty,
+				TypedBinding.ForSingleNestingLevel(
+					nameof(ShellSection.FlyoutDisplayOptions),
+					static (ShellSection section) => section.FlyoutDisplayOptions,
+					mode: BindingMode.OneTime,
```

Comment:
This binding has the `OneTime` mode while the other ones have `OneWay`. Is that correct?